### ISSUE TITLE
Create TLS secret from AWS Secret Manager

### ISSUE THREAD

**Issue by ghost** [external-secrets/external-secrets#2774]
Hi everyone,

I don't know really if it is a bug or just a release that doesn't exist but my problem is : 

I would like to generate a TLS type secret from aws secret manager.

Firstly, i generated my pkcs12 : 
openssl pkcs12 -export -inkey server.key -in server.crt -passout pass: -out server.p12
Then base64 server.p12 > serverbase64.txt

I copied the base64 content to my aws secret :
![image](https://github.com/external-secrets/external-secrets/assets/147515761/0c2377c7-f0ba-4f9a-b6a4-15438e18d64a)

And finally i tried to create the external secret like that : 

```
apiVersion: external-secrets.io/v1beta1 
kind: ExternalSecret 
metadata: 
  name: mycrt 
  namespace: ... 
spec: 
  data: 
  - remoteRef: 
      conversionStrategy: Default 
      decodingStrategy: Base64 
      key: path/of/secret/aws 
    secretKey: mycrt 
  refreshInterval: 1s 
  secretStoreRef: 
    kind: ClusterSecretStore 
    name: clustersecretstore-aws 
  target: 
    creationPolicy: Owner 
    deletionPolicy: Retain 
    name: mycrt 
    template: 
      data: 
        tls.crt: '{{ .mycrt | pkcs12cert }}' 
        tls.key: '{{ .mycrt| pkcs12key }}' 
      engineVersion: v2 
      mergePolicy: Replace 
      type: kubernetes.io/tls 
```

I refered to that documentation : https://external-secrets.io/latest/guides/templating/ (with the TLS part) for the kubernetes part operator.
I tried to tls.crt: '{{ .mycrt | pkcs12certPass "mypassword" }}' that i put when prompt for the pkcs command

The result stay unsynchronized, and display : 

`could not apply template: could not execute template: could not execute template: unable to execute template at key tls.crt: unable to execute template at key tls.crt: template: tls.crt:1:17: executing "tls.crt" at <pkcs12certPass "mypassword">: error calling pkcs12certPass: unable to decode pkcs12 certificate with password: pkcs12: unknown digest algorithm: 2.16.840.1.101.3.4.2.1

I don't know if i did something wrong.

Version helm operator : 0.9.1

Thanks to you if you can help me !

Best regards

**Comment by moolen** [comment]
you first have to `base64 decode` the secret content in the template before doing `pkcs12cert`.

```
    template: 
      data: 
        tls.crt: '{{ .mycrt | b64dec | pkcs12cert }}' 

```

**Comment by Hyz90** [comment]
> b64dec

Thanks for your answer.
On my side it didn't work, there is already the base64 code strategy, so it decoded twice.

So i did it differently, i just copied the private key and the certificate together in the secret aws and encode it in base64.

In the manifest of the external secret i did it with the filter PEM :

```
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: mycrt
  namespace: ...
spec:
  data:
  - remoteRef:
      conversionStrategy: Default
      decodingStrategy: Base64
      key: path/of/key/aws
    secretKey: mycrt
  refreshInterval: 1h
  secretStoreRef:
    kind: ClusterSecretStore
    name: clustersecretstore-aws
  target:
    creationPolicy: Owner
    deletionPolicy: Retain
    name: mycrt
    template:
      data:
        tls.crt: '{{ .mycrt| filterPEM "CERTIFICATE" }}'
        tls.key: '{{ .mycrt | filterPEM "PRIVATE KEY"}}'
      engineVersion: v2
      mergePolicy: Replace
      type: kubernetes.io/tls
```
Have a good day !

**Comment by lhaussknecht** [comment]
So basicly you avoided PKCS#12 altogether (which is a nice workaround btw).

The underlying problem with PKCS#12 seems to be go's inability in parsing SHA256 HMACs from newer openssl versions: https://github.com/golang/go/issues/62375.

Hashicorp referenced another package to be able to parse SHA256 HMACs: https://github.com/hashicorp/go-azure-sdk/pull/328

Should we wait for the go package to be updated or do it like the Hashicorp folks?

**Comment by gusfcarvalho** [comment]
The guideline is any license as permissive or more as Apache 2. That library seems to comply to it as it is BSD3. I’m just not sure if we may run into compatibility issues when we go for ubi-boringssl builds though 🤔

**Comment by lhaussknecht** [comment]
Good point. I have no clue.

In the meantime I created a PKCS#12 certificate on RHEL 8.9 with `openSSL 1.1.1k  FIPS 25 Mar 2021` and it works with the pkcs12cert template. `OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)` on my Ubunut machine creates PKCS#12 with SHA256, that cannot be processed by the template.

Maybe we can add a hint in the readme to either use an older version of openssl (which version exactly?) or use the workaround by using `filterPEM` if you receive the `unknown digest algorithm: 2.16.840.1.101.3.4.2.1` error.

**Comment by saltbo** [comment]
> Good point. I have no clue.
> 
> In the meantime I created a PKCS#12 certificate on RHEL 8.9 with `openSSL 1.1.1k FIPS 25 Mar 2021` and it works with the pkcs12cert template. `OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)` on my Ubunut machine creates PKCS#12 with SHA256, that cannot be processed by the template.
> 
> Maybe we can add a hint in the readme to either use an older version of openssl (which version exactly?) or use the workaround by using `filterPEM` if you receive the `unknown digest algorithm: 2.16.840.1.101.3.4.2.1` error.

Good idea, any progress to update the document? we need it!

**Comment by saltbo** [comment]
https://github.com/external-secrets/external-secrets/pull/3601

I found this PR has fixed this issue in v0.9.20
